packages/components/src/DynamicDisclosure/DynamicDisclosure.tsx

```diff
@@ -19,13 +19,18 @@ interface DynamicDisclosureContextType {
   onClose: () => void;
   isOpen: boolean;
   goBack: () => void;
+  stack: Array<{
```

Comment:
since the only thing you need this for is to check if you need to show a back button, let's have a boolean `hasPreviousPage` or `canGoBack` or whatever. this stack is a too internal thing, and having the same condition implemented in all places we might have a back button is tedious